import/controls/TextSelectionToolbar.qml

```diff
@@ -0,0 +1,100 @@
+/****************************************************************************
+**
+** Copyright (c) 2016 - 2020 Jolla Ltd.
+** Copyright (c) 2020 Open Mobile Platform LLC.
+**
+****************************************************************************/
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import QtQuick 2.6
+import Sailfish.Silica 1.0
+import Sailfish.WebView.Controls 1.0
+
+Row {
+    id: root
+
+    property TextSelectionController controller
+
+    property bool isPhoneNumber: controller && controller.isPhoneNumber
+    property bool hasSearch: controller && controller.searchUri !== ""
+    property bool selectAllEnabled
+
+    property real _buttonWidth: width / (repeater.count + 1)
+
+    readonly property var defaultButtons: {
+        var buttons = []
+
+        if (isPhoneNumber) {
+            buttons.push({
+                "icon": "image://theme/icon-m-answer",
+                "action": function () {
+                    call()
+                    clear()
+                }
+            })
+        }
+        buttons.push({
+            "icon": "image://theme/icon-m-share",
+            "action": function () {
+                share()
+                clear()
+            }
+        })
+
+        if (selectAllEnabled) {
+            buttons.push({ "icon": "image://theme/icon-m-select-all", "action": selectAll })
+        }
+
+        if (hasSearch) {
+            buttons.push({
+                "icon": "image://theme/icon-m-search",
+                "action": function () {
+                    search()
+                    clear()
```

Comment:
Regarding the review comment. It's better to first clear the search text and then trigger the action. Same applies for the other actions.

Reply:
But then the text wouldn't be available for search.

Reply:
Browser side was previously like this:

                            controller.clearSelection()
                            webView.tabModel.newTab(controller.searchUri)
                            overlay.animator.showChrome(true)

Clearing selection did not clear yet the related search information.

Reply:
I've reversed the order so the clear() happens first. Irrespective of order though it was noted in another PR that clearing them immediately makes the icons change while a share page is being pushed so there's a little friction there.